R/mobr.R

```diff
@@ -10,221 +10,244 @@ require(pracma)
 ## sample-based implicit, and individual-based rescaled to sample
 get_avg_dens = function(dat_sp, dat_plot, ScaleBy) {
     # Auxillary function to obtain average density within plot for rescaling Ask user
-    # to specify which treatment is used as 'standard' for rescaling.
+    # to specify which group is used as 'standard' for rescaling.
     if (!is.na(ScaleBy)) {
-        dat_scale = dat_sp[match(dat_plot[which(dat_plot[ , 2] == ScaleBy), 1],
-                                 dat_sp[ , 1]), 2:ncol(dat_sp)]
-        avg_dens = mean(apply(dat_scale, 1, sum))
+        avg_dens = mean(apply(dat_sp[dat_plot$group == ScaleBy, ], 1, sum))
```

Comment:
This does assume that they've already been matched. My suggestion for checking that the sites names match is to either have a function that can be called that we run at the beginning of all these different comparison functions or we just tell the user to make sure that their rows in their two matrices match. For example the package `vegan` does not check that site names match between community and explanatory matrices.